packages/typespec-test/eng/smoke-test.js

```diff
@@ -8,6 +8,9 @@ const MAX_BUFFER = 10 * 1024 * 1024;
 function generate(path) {
   let command = `cd ${path} && npx tsp compile ./spec`;
   try {
+    if (existsSync(join(path, "spec", "tspconfig.yaml"))) {
+      command = `cd ${path}/spec && npx tsp compile --config ./tspconfig.yaml .`
```

Comment:
Then we will lose the entry point selection - the client.tsp or main.tsp

Reply:
the append client.tsp logic is in the below part.